dwave/plugins/sklearn/transformers.py

```diff
@@ -311,3 +408,167 @@ def fit(
+            Whether to compute the off-diagonal terms using the conditional mutual
+            information or joint mutual information
+
+        discrete_features: See :func:`sklearn.feature_selection._mutual_info._estimate_mi`
+        
+        discrete_target: See :func:`sklearn.feature_selection._mutual_info._estimate_mi`
+        
+        n_neighbors: See :func:`sklearn.feature_selection._mutual_info._estimate_mi`
+        
+        copy: See :func:`sklearn.feature_selection._mutual_info._estimate_mi`
+        
+        random_state: See :func:`sklearn.feature_selection._mutual_info._estimate_mi`
+
+        n_jobs: int, default=None
+            The number of parallel jobs to run for the conditional mutual information
+            computation. The parallelization is over the columns of `X`.
+            ``None`` means 1 unless in a :obj:`joblib.parallel_backend` context.
+            ``-1`` means using all processors. See :term:`Glossary <n_jobs>`
+            for more details.
+        
+    Returns:
+        mi_matrix : ndarray, shape (n_features, n_features)
+        Interaction matrix between the features using (conditional) mutual information.
+        A negative value will be replaced by 0.
+
+    References:    
+    .. [1] A. Kraskov, H. Stogbauer and P. Grassberger, "Estimating mutual
+           information". Phys. Rev. E 69, 2004.
+    .. [2] B. C. Ross "Mutual Information between Discrete and Continuous
+           Data Sets". PLoS ONE 9(2), 2014.
+    .. [3] Mesner, Octavio César, and Cosma Rohilla Shalizi. "Conditional
+           mutual information estimation for mixed, discrete and continuous
+           data." IEEE Transactions on Information Theory 67.1 (2020): 464-484.
+    """
+    X, y = check_X_y(X, y, accept_sparse="csc", y_numeric=not discrete_target)
+    n_samples, n_features = X.shape
+    
+    if isinstance(discrete_features, (str, bool)):
+        if isinstance(discrete_features, str):
+            if discrete_features == "auto":
+                discrete_features = issparse(X)
+            else:
+                raise ValueError("Invalid string value for discrete_features.")
+        discrete_mask = np.empty(n_features, dtype=bool)
+        discrete_mask.fill(discrete_features)
+    else:
+        discrete_features = check_array(discrete_features, ensure_2d=False)
+        if np.issubdtype(discrete_features.dtype, bool):
+            discrete_mask = np.zeros(n_features, dtype=bool)
+            discrete_mask[discrete_features] = True
+        else:
+            discrete_mask = discrete_features
+
+    # copying X if needed
+    if copy:
+        X = X.copy()
+  
+    # Computing the diagonal terms
+    mutual_info_args = dict(X=X, y=y,
+                              discrete_features=discrete_mask,
+                              n_neighbors=n_neighbors,
+                              copy=False, random_state=random_state)
+    # In earlier versions of sklearn functions `mutual_info_classif` and 
```

Comment:
What version of scikit-learn did the parameters we want come in? I have no issue just raising the lower bound to save on switching behavior like this.

Reply:
It hasn't been released yet. They said they'll release in May